### PR TITLE
chore: Improve wording of error message

### PR DESCRIPTION
**Pull request by WanYixian**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

According to [request](https://risingwave-labs.slack.com/archives/C0358PVT2AX/p1719544129504999), use `bail!` `.context` `anyhow` `Err` `String.format` to do a global search of error messages and improve the wording.

This is the [guidance](https://www.notion.so/risingwave-labs/Improving-RisingWave-error-messages-9e4ffd5142a840c1bfcd8e2ef32ca3b6).

Feel free to check out and see if all changes make sense, thanks!

### CODE REVIEW

File: e2e_test/source/basic/old_row_format_syntax/ddl.slt [hunk 1]
```diff
@@ -7,7 +7,7 @@ create source s (
   properties.bootstrap.server = 'message_queue:29092'
 ) ROW FORMAT JSON;
 
-statement error properties `scan_startup_mode` only support earliest and latest or leave it empty
+statement error properties `scan_startup_mode` only supports earliest and latest or leaving it empty
```
Comment (ShanlanLi) [inline e2e_test/source/basic/old_row_format_syntax/ddl.slt:10]:
```suggestion
statement error property `scan_startup_mode` only accepts three options: earliest, latest, or left empty
```

File: src/stream/src/executor/dynamic_filter.rs [hunk 2]
```diff
@@ -403,8 +403,8 @@ impl<S: StateStore, const USE_WATERMARK_CACHE: bool> DynamicFilterExecutor<S, US
                         let (range, _latest_is_lower, is_insert) = self.get_range(&curr, prev);
 
                         if !is_insert && self.condition_always_relax {
-                            bail!("The optimizer inferred that the right side's change always make the condition more relaxed.\
-                                But the right changes make the conditions stricter.");
+                            bail!("The optimizer incorrectly assumed that changes on the right side always relax the condition.\
```
Comment (ShanlanLi) [inline src/stream/src/executor/dynamic_filter.rs:406]: It's a bit hard to understand this message.

**Review by ShanlanLi — APPROVED** [review]
LGTM

File: java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java [hunk 3]
```diff
@@ -91,7 +91,7 @@ public static void checkSchema(
                 throw Status.FAILED_PRECONDITION
                         .withDescription(
                                 String.format(
-                                        "Don't match in the name, rw is %s cassandra can't find it",
+                                        "Name mismatch. %s of RisingWave is not found in Cassandra.",
```
Comment (stdrc) [inline java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java:94]:
```suggestion
                                        "Name mismatch. Column `%s` on RisingWave side is not found on Cassandra side.",
```

File: java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java [hunk 4]
```diff
@@ -100,7 +100,7 @@ public static void checkSchema(
                 throw Status.FAILED_PRECONDITION
                         .withDescription(
                                 String.format(
-                                        "Don't match in the type, name is %s, cassandra is %s, rw is %s",
+                                        "Type mismatch, name is %s, cassandra is %s, rw is %s",
```
Comment (stdrc) [inline java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java:103]:
```suggestion
                                        "Data type mismatch for column `%s`. Cassandra side: `%s`, RisingWave side: `%s`.",
```

File: java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java [hunk 5]
```diff
@@ -113,7 +113,7 @@ public static void checkPrimaryKey(
             List<ColumnMetadata> cassandraColumnMetadatas, List<String> columnMetadatas) {
         if (cassandraColumnMetadatas.size() != columnMetadatas.size()) {
             throw Status.FAILED_PRECONDITION
-                    .withDescription("Primary key len don't match")
+                    .withDescription("Primary key len mismatch")
```
Comment (stdrc) [inline java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java:116]:
```suggestion
                    .withDescription("Primary key length mismatch.")
```

File: java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java [hunk 6]
```diff
@@ -125,7 +125,7 @@ public static void checkPrimaryKey(
                 throw Status.FAILED_PRECONDITION
                         .withDescription(
                                 String.format(
-                                        "Primary key don't match. RisingWave Primary key is %s, don't find it in cassandra",
+                                        "Primary key mismatch. RisingWave Primary key is %s, not found in cassandra",
```
Comment (stdrc) [inline java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java:128]:
```suggestion
                                        "Primary key mismatch. Primary key `%s` on RisingWave side is not found on Cassandra side",
```

File: java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-common/src/main/java/com/risingwave/mock/flink/common/FlinkDynamicUtil.java [hunk 7]
```diff
@@ -120,14 +120,14 @@ public static void validateSchemaWithCatalog(
                 throw Status.FAILED_PRECONDITION
                         .withDescription(
                                 String.format(
-                                        "Don't match in the name, rw is %s", columnDesc.getName()))
+                                        "Name mismatch. RisingWave is %s", columnDesc.getName()))
```
Comment (stdrc) [inline java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-common/src/main/java/com/risingwave/mock/flink/common/FlinkDynamicUtil.java:123]:
```suggestion
                                        "Name mismatch. Column `%s` on RisingWave side is not found on Flink side.", columnDesc.getName()))
```

File: java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-common/src/main/java/com/risingwave/mock/flink/common/FlinkDynamicUtil.java [hunk 8]
```diff
@@ -120,14 +120,14 @@ public static void validateSchemaWithCatalog(
...
                         .asRuntimeException();
             }
             if (!checkType(columnDesc.getDataType(), flinkColumnMap.get(columnDesc.getName()))) {
                 throw Status.FAILED_PRECONDITION
                         .withDescription(
                                 String.format(
-                                        "Don't match in the type, name is %s, Sink is %s, rw is %s",
+                                        "Type mismatch. Name is %s, Sink is %s, RisingWave is %s",
```
Comment (stdrc) [inline java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-common/src/main/java/com/risingwave/mock/flink/common/FlinkDynamicUtil.java:130]:
```suggestion
                                        "Data type mismatch for column `%s`. Flink side: `%s`, RisingWave side: `%s`.",
```

File: src/connector/src/sink/big_query.rs [hunk 9]
```diff
@@ -211,14 +211,14 @@ impl BigQuerySink {
         for i in rw_fields_name {
             let value = big_query_columns_desc.get(&i.name).ok_or_else(|| {
                 SinkError::BigQuery(anyhow::anyhow!(
-                    "Column name don't find in bigquery, risingwave is {:?} ",
+                    "Column name is not found in bigquery, risingwave is {:?} ",
```
Comment (stdrc) [inline src/connector/src/sink/big_query.rs:214]:
```suggestion
                    "Column `{:?}` on RisingWave side is not found on BigQuery side.",
```

File: src/connector/src/sink/big_query.rs [hunk 10]
```diff
@@ -211,14 +211,14 @@ impl BigQuerySink {
...
                     i.name
                 ))
             })?;
             let data_type_string = Self::get_string_and_check_support_from_datatype(&i.data_type)?;
             if data_type_string.ne(value) {
                 return Err(SinkError::BigQuery(anyhow::anyhow!(
-                    "Column type don't match, column name is {:?}. bigquery type is {:?} risingwave type is {:?} ",i.name,value,data_type_string
+                    "Column type mismatch, column name is {:?}. bigquery type is {:?} risingwave type is {:?} ",i.name,value,data_type_string
```
Comment (stdrc) [inline src/connector/src/sink/big_query.rs:221]:
```suggestion
                    "Data type mismatch for column `{:?}`. BigQuery side: `{:?}`, RisingWave side: `{:?}`. ", i.name, value, data_type_string
```

File: java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java (hunk 4 above)
Comment (stdrc) [reply java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraUtil.java:103]: Here `Cassandra` is a proper noun, just like `RisingWave`.

**Review by stdrc — COMMENTED** [review]
Added some change suggestions, including the following several aspects:

1. Names (of columns, keys, schemas, tables, etc.) should be in `` ` `` inline code.
2. Proper noun should be in proper form, e.g. `Flink`, `RisingWave`, `BigQuery`.

Another suggestion: Maybe we should prefer passive voice like `Column (is) not found`, `Xxx mode is not supported`, ``Property `xxx` of Xxx is not supported ``, `Xxx required by xxx is not found`. In this way, we can emphasize the object that is problematic.

File: src/connector/src/sink/deltalake.rs [hunk 11]
```diff
@@ -356,7 +356,7 @@ impl Sink for DeltaLakeSink {
             .collect();
         if deltalake_fields.len() != self.param.schema().fields().len() {
             return Err(SinkError::DeltaLake(anyhow!(
-                "column count not match, rw is {}, deltalake is {}",
+                "column mismatch, risingwave is {}, deltalake is {}",
```
Comment (fuyufjh) [inline src/connector/src/sink/deltalake.rs:359]:
```suggestion
                "Columns mismatch. RisingWave is {}, DeltaLake is {}",
```

File: src/connector/src/sink/iceberg/mod.rs [hunk 12]
```diff
@@ -1055,7 +1055,7 @@ impl WriteResult {
                     .collect::<std::result::Result<Vec<DataFile>, icelake::Error>>()
                     .context("Failed to parse data file from json")?;
             } else {
-                bail!("icberg sink metadata should have data_files object");
+                bail!("iceberg sink metadata should have data_files object");
```
Comment (fuyufjh) [inline src/connector/src/sink/iceberg/mod.rs:1058]:
```suggestion
                bail!("Iceberg sink metadata should have data_files object");
```

File: src/connector/src/sink/mqtt.rs [hunk 13]
```diff
@@ -133,7 +133,7 @@ impl MqttConfig {
             .map_err(|e| SinkError::Config(anyhow!(e)))?;
         if config.r#type != SINK_TYPE_APPEND_ONLY {
             Err(SinkError::Config(anyhow!(
-                "Mqtt sink only support append-only mode"
+                "Mqtt sink only supports append-only mode"
```
Comment (fuyufjh) [inline src/connector/src/sink/mqtt.rs:136]:
```suggestion
                "MQTT sink only supports append-only mode"
```

File: src/connector/src/sink/mqtt.rs [hunk 14]
```diff
@@ -175,7 +175,7 @@ impl Sink for MqttSink {
     async fn validate(&self) -> Result<()> {
         if !self.is_append_only {
             return Err(SinkError::Mqtt(anyhow!(
-                "Mqtt sink only support append-only mode"
+                "Mqtt sink only supports append-only mode"
```
Comment (fuyufjh) [inline src/connector/src/sink/mqtt.rs:178]:
```suggestion
                "MQTT sink only supports append-only mode"
```

File: src/connector/src/sink/mqtt.rs [hunk 15]
```diff
@@ -261,7 +261,7 @@ impl MqttSinkWriter {
             },
             _ => {
                 return Err(SinkError::Config(anyhow!(
-                    "Mqtt sink only support append-only mode"
+                    "Mqtt sink only supports append-only mode"
```
Comment (fuyufjh) [inline src/connector/src/sink/mqtt.rs:264]:
```suggestion
                    "MQTT sink only supports append-only mode"
```

File: src/connector/src/sink/nats.rs [hunk 16]
```diff
@@ -79,7 +79,7 @@ impl NatsConfig {
             .map_err(|e| SinkError::Config(anyhow!(e)))?;
         if config.r#type != SINK_TYPE_APPEND_ONLY {
             Err(SinkError::Config(anyhow!(
-                "Nats sink only support append-only mode"
+                "Nats sink only supports append-only mode"
```
Comment (fuyufjh) [inline src/connector/src/sink/nats.rs:82]:
```suggestion
                "NATS sink only supports append-only mode"
```

File: src/connector/src/sink/nats.rs [hunk 17]
```diff
@@ -117,7 +117,7 @@ impl Sink for NatsSink {
     async fn validate(&self) -> Result<()> {
         if !self.is_append_only {
             return Err(SinkError::Nats(anyhow!(
-                "Nats sink only support append-only mode"
+                "Nats sink only supports append-only mode"
```
Comment (fuyufjh) [inline src/connector/src/sink/nats.rs:120]:
```suggestion
                "NATS sink only supports append-only mode"
```

**Review by fuyufjh — APPROVED** [review]
LGTM.

**Comment by stdrc** [comment]
@BugenZhao any other suggestions?

**Pull request merged**